src/components/data/_tests/data-utils.test.js

```diff
@@ -61,3 +61,79 @@ describe('concatenateAttributeNames', () => {
     expect(concatenateAttributeNames(attrList1, attrList2)).toEqual(expected)
   })
 })
+
+describe('getAttributeType', () => {
+  it('returns type of attribute value', () => {
+    expect(getAttributeType('value')).toEqual({ type: 'string', isList: false })
+    expect(getAttributeType(3)).toEqual({ type: 'number', isList: false })
+    expect(getAttributeType(false)).toEqual({ type: 'boolean', isList: false })
+    expect(getAttributeType({
+      entityType: 'thing',
+      entityName: 'thing_one'
+    })).toEqual({ type: 'reference', isList: false })
+
+    expect(getAttributeType({ items: ['a', 'b', 'c'] })).toEqual({ type: 'string', isList: true })
+    expect(getAttributeType({ items: [1, 2, 3] })).toEqual({ type: 'number', isList: true })
+    expect(getAttributeType({ items: [true, false] })).toEqual({ type: 'boolean', isList: true })
+    expect(getAttributeType({
+      items: [
+        { entityType: 'thing', entityName: 'thing_one' },
+        { entityType: 'thing', entityName: 'thing_two' }
+      ],
+      itemsType: 'EntityReference'
+    })).toEqual({ type: 'reference', isList: true })
+  })
+
+  it('returns string for null values', () => {
+    expect(getAttributeType(null)).toEqual({ type: 'string', isList: false })
```

Comment:
These tests are great! Some of these conversions are not at all obvious, and it is great to have documentation of what they do.